### PR TITLE
Remove use of push descriptor extension from graphics_pipeline benchmark

### PR DESCRIPTION
**Pull request by angela28chen**
* Replace push descriptors with traditional descriptors
* Rename variables with `Skybox` -> `SkyBox` for consistency
* Some comment/spacing corrections for bigwheels (not functional)
* Minor adjustments to knob appearance in UI (indent, ordering)

### CODE REVIEW

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 1]
```diff
@@ -141,10 +141,19 @@ void GraphicsBenchmarkApp::Setup()
         samplerCreateInfo.maxLod                  = FLT_MAX;
         PPX_CHECKED_CALL(GetDevice()->CreateSampler(&samplerCreateInfo, &mLinearSampler));
     }
+    // Descriptor Pool
+    {
+        grfx::DescriptorPoolCreateInfo createInfo = {};
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:146]: why do you need 100 for each?

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 2]
```diff
@@ -302,18 +326,72 @@ void GraphicsBenchmarkApp::SetupFullscreenQuadsResources()
...
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateSampler(SPHERE_METAL_ROUGHNESS_SAMPLER_REGISTER, 0, mLinearSampler));
+
+        if (pAllTexturesTo1x1->GetValue()) {
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_ALBEDO_SAMPLED_IMAGE_REGISTER, 0, mWhitePixelTexture));
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_NORMAL_SAMPLED_IMAGE_REGISTER, 0, mWhitePixelTexture));
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_METAL_ROUGHNESS_SAMPLED_IMAGE_REGISTER, 0, mWhitePixelTexture));
+        }
+        else {
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_ALBEDO_SAMPLED_IMAGE_REGISTER, 0, mAlbedoTexture));
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_NORMAL_SAMPLED_IMAGE_REGISTER, 0, mAlbedoTexture));
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:379]: mNormalMapTexture

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 3]
```diff
@@ -302,18 +326,72 @@ void GraphicsBenchmarkApp::SetupFullscreenQuadsResources()
...
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_METAL_ROUGHNESS_SAMPLED_IMAGE_REGISTER, 0, mAlbedoTexture));
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:380]: mMetalRoughnessTexture

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 4]
```diff
@@ -847,6 +940,12 @@ void GraphicsBenchmarkApp::RecordCommandBuffer(PerFrame& frame, grfx::SwapchainP
         frame.cmd->BindGraphicsPipeline(mQuadsPipelines[pFullscreenQuadsType->GetIndex()]);
         frame.cmd->BindVertexBuffers(1, &mFullscreenQuads.vertexBuffer, &mFullscreenQuads.vertexBinding.GetStride());
 
+        if (pFullscreenQuadsType->GetIndex() == static_cast<size_t>(FullscreenQuadsType::FULLSCREEN_QUADS_TYPE_TEXTURE)) {
+            grfx::DescriptorSet* sets[1] = {nullptr};
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:944]: i guess we could put `mFullscreenQuads.descriptorSets.at(GetInFlightFrameIndex()` directly in the initialize list? or maybe no need to create this array, just pass `&mFullscreenQuads.descriptorSets.at(GetInFlightFrameIndex())` directly?

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 5]
```diff
@@ -899,31 +998,38 @@ void GraphicsBenchmarkApp::RecordCommandBuffer(PerFrame& frame, grfx::SwapchainP
...
-        data.MVP        = frame.sceneData.viewProjectionMatrix * glm::scale(float3(500.0f, 500.0f, 500.0f));
-        mSkyBox.uniformBuffer->CopyFromSource(sizeof(data), &data);
 
-        frame.cmd->PushGraphicsUniformBuffer(mSkyBox.pipelineInterface, /* binding = */ 0, /* set = */ 0, /* bufferOffset = */ 0, mSkyBox.uniformBuffer);
-        frame.cmd->PushGraphicsSampledImage(mSkyBox.pipelineInterface, /* binding = */ 1, /* set = */ 0, mSkyBoxTexture->GetSampledImageView());
-        frame.cmd->PushGraphicsSampler(mSkyBox.pipelineInterface, /* binding = */ 2, /* set = */ 0, mLinearSampler);
-    }
+    grfx::DescriptorSet* sets[1] = {nullptr};
+    sets[0]                      = mSkyBox.descriptorSets.at(GetInFlightFrameIndex());
+    frame.cmd->BindGraphicsDescriptorSets(mSkyBox.pipelineInterface, 1, sets);
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:1010]: same as above

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp [hunk 6]
```diff
@@ -899,31 +998,38 @@ void GraphicsBenchmarkApp::RecordCommandBuffer(PerFrame& frame, grfx::SwapchainP
...
 void GraphicsBenchmarkApp::RecordCommandBufferSpheres(PerFrame& frame)
 {
+    // Bind resources
     const size_t pipelineIndex = mGraphicsPipelinesIndexer.GetIndex({pKnobVs->GetIndex(), pKnobPs->GetIndex(), pKnobVbFormat->GetIndex(), pKnobVertexAttrLayout->GetIndex()});
     frame.cmd->BindGraphicsPipeline(mPipelines[pipelineIndex]);
     const size_t meshIndex = mMeshesIndexer.GetIndex({pKnobLOD->GetIndex(), pKnobVbFormat->GetIndex(), pKnobVertexAttrLayout->GetIndex()});
     frame.cmd->BindIndexBuffer(mSphereMeshes[meshIndex]);
     frame.cmd->BindVertexBuffers(mSphereMeshes[meshIndex]);
 
+    grfx::DescriptorSet* sets[1] = {nullptr};
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:1029]: same as above

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.h [hunk 7]
```diff
@@ -36,6 +36,20 @@ static constexpr uint32_t kMaxSphereInstanceCount  = 3000;
 static constexpr uint32_t kSeed                    = 89977;
 static constexpr uint32_t kMaxFullscreenQuadsCount = 1000;
 
+static constexpr size_t SKYBOX_UNIFORM_BUFFER_REGISTER = 0;
```
Comment (wangra-google) [inline benchmarks/graphics_pipeline/GraphicsBenchmarkApp.h:39]: do they need to be in the header?

**Review by wangra-google — CHANGES_REQUESTED** [review]

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp (hunk 1 above)
Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:160]: Reduced to 10

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp (hunk 2 above)
Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp]: Oops, thanks for catching this. I was trying to figure out why the spheres looked different.

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp (hunk 3 above)
Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp]: Fixed

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp (hunk 4 above)
Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp]: Sure, changed this and other cases

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.h (hunk 7 above)
Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.h]: Moved to cpp

File: benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp (hunk 1 above)
Comment (wangra-google) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:160]: sorry still a bit confused, why is it 10? according to the comments, should sampler be 4, sampledImage be 5 and uniformbuffer be 2?

Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:161]: I thought it was acceptable to allocate a bit extra, also I wasn't sure about if # of frames in flight would change. But I see now that it's a setting so I've added that explicitly in `Config` and I've reduced the descriptor counts to 4/5/2.

Comment (wangra-google) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:161]: or would it be better to let each one multiply by GetNumFramesInFlight()?

Comment (angela28chen) [reply benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp:161]: Sounds good, fixed

**Review by wangra-google — APPROVED** [review]

**Review by apazylbe — APPROVED** [review]

**Comment by angela28chen** [comment]
Merged the sphere enable knob, made some minor adjustments to how the knobs are presented in the UI

**Pull request merged**